workspace.spec

```diff
@@ -572,6 +572,359 @@ module Workspace {
+
+		If the contributor is an organization, either the 'contributor_id' or
+		the 'name' field must be populated.
+
+		An individual can be represented by either 'given_name' and 'family_name'; a 'name'
+		(if the name is not in the binomial format); or a `contributor_id`, such as an
+		ORCID.
+
+		The 'contributor_role' field takes values from the Datacite and CRediT contributor
+		roles vocabularies. For more information on these resources and choosing the
+		appropriate roles, please see the following links:
+
+		Datacite contributor roles: https://support.datacite.org/docs/datacite-metadata-schema-v44-recommended-and-optional-properties#7a-contributortype
+
+		CRediT contributor role taxonomy: https://credit.niso.org
+
+		string contributor_type - must be either 'person' or 'organization'
+		string contributor_id (optional) - persistent unique identifier for the contributor;
+			e.g. an ORCID for an individual, or a ROR ID for an organization.
+		string given_name (optional) - first name, for individuals whose names can be split
+			into given and surnames
+		string family_name (optional) - family name, for individuals whose names can be
+			split into given and surnames.
+		string name (optional) - contributor name, in cases where the contributor is an
+			organization or a person whose name cannot easily fit into the given/family
+			name structure.
+		list<Organization> affiliations (optional) - list of organizations with which the
+			contributor is affiliated
+		list<string> contributor_roles (optional) - list of roles played by the contributor
+			when working on the resource.
+			Valid contributor_role values:
+			"Datacite:ContactPerson"
+			"Datacite:DataCollector"
+			"Datacite:DataCurator"
+			"Datacite:DataManager"
+			"Datacite:Distributor"
+			"Datacite:Editor"
+			"Datacite:HostingInstitution"
+			"Datacite:Producer"
+			"Datacite:ProjectLeader"
+			"Datacite:ProjectManager"
+			"Datacite:ProjectMember"
+			"Datacite:RegistrationAgency"
+			"Datacite:RegistrationAuthority"
+			"Datacite:RelatedPerson"
+			"Datacite:Researcher"
+			"Datacite:ResearchGroup"
+			"Datacite:RightsHolder"
+			"Datacite:Sponsor"
+			"Datacite:Supervisor"
+			"Datacite:WorkPackageLeader"
+			"Datacite:Other"
+			"CRediT:conceptualization"
+			"CRediT:data-curation"
+			"CRediT:formal-analysis"
+			"CRediT:funding-acquisition"
+			"CRediT:investigation"
+			"CRediT:methodology"
+			"CRediT:project-administration"
+			"CRediT:resources"
+			"CRediT:software"
+			"CRediT:supervision"
+			"CRediT:validation"
+			"CRediT:visualization"
+			"CRediT:writing-original-draft"
+			"CRediT:writing-review-editing"
+	*/
+	typedef structure {
+		string contributor_type;
+		string name;
+		string given_name;
+		string family_name;
+		string contributor_id;
+		list<Organization> affiliations;
+		list<string> contributor_roles;
+	} Contributor;
+
+	/*
+		Represents the credit metadata associated with a workspace object.
```

Comment:
not yet added

Reply:
This looks good. At the minimum should cover the fields to display the recommended ESIP citation, in case we want to add that to documentation.
https://esip.figshare.com/articles/online_resource/Data_Citation_Guidelines_for_Earth_Science_Data_Version_2/8441816